src/Illuminate/Auth/VerifyEmails/DatabaseTokenRepository.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace Illuminate\Auth\VerifyEmails;
+
+use Carbon\Carbon;
+use Illuminate\Support\Str;
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Contracts\Auth\CanVerifyEmail as CanVerifyEmailContract;
+
+class DatabaseTokenRepository implements TokenRepositoryInterface
+{
+    /**
+     * The database connection instance.
+     *
+     * @var \Illuminate\Database\ConnectionInterface
+     */
+    protected $connection;
+
+    /**
+     * The token database table.
+     *
+     * @var string
+     */
+    protected $table;
+
+    /**
+     * The hashing key.
+     *
+     * @var string
+     */
+    protected $hashKey;
+
+    /**
+     * The number of seconds a token should last.
+     *
+     * @var int
+     */
+    protected $expires;
+
+    /**
+     * Create a new token repository instance.
+     *
+     * @param  \Illuminate\Database\ConnectionInterface  $connection
+     * @param  string  $table
+     * @param  string  $hashKey
+     * @param  int  $expires
+     * @return void
+     */
+    public function __construct(ConnectionInterface $connection, $table, $hashKey, $expires = 60)
+    {
+        $this->table = $table;
+        $this->hashKey = $hashKey;
+        $this->expires = $expires * 60;
+        $this->connection = $connection;
+    }
+
+    /**
+     * Create a new token record.
+     *
+     * @param  \Illuminate\Contracts\Auth\CanVerifyEmail  $user
+     * @return string
+     */
+    public function create(CanVerifyEmailContract $user)
+    {
+        $email = $user->getEmailToVerify();
+
+        $this->deleteExisting($user);
+
+        // We will create a new, random token for the user so that we can e-mail them
+        // a safe link to verify their email address. Then we will insert a record in
+        // the database so that we can verify the token later.
+        $token = $this->createNewToken();
+
+        $this->getTable()->insert($this->getPayload($email, $token));
+
+        return $token;
+    }
+
+    /**
+     * Delete all existing tokens from the database.
+     *
+     * @param  \Illuminate\Contracts\Auth\CanVerifyEmail  $user
+     * @return int
+     */
+    protected function deleteExisting(CanVerifyEmailContract $user)
+    {
+        return $this->getTable()->where('email', $user->getEmailToVerify())->delete();
+    }
+
+    /**
+     * Build the record payload for the table.
+     *
+     * @param  string  $email
+     * @param  string  $token
+     * @return array
+     */
+    protected function getPayload($email, $token)
+    {
+        return ['email' => $email, 'token' => $token, 'created_at' => new Carbon];
```

Comment:
Leaving this one for now, since it's the same situation in Passwords\DatabaseTokenRepository.